tests/core/persistent-journal-namespace/task.yaml

```diff
@@ -0,0 +1,61 @@
+summary: Test journal.persistent core config option is honored by journal namespaces
+
+# these systems do not support journal quota groups due to their old systemd versions.
+# requires systemd v245+
+systems:
+  - -ubuntu-core-16-*
+  - -ubuntu-core-18-*
+
+prepare: |
+  systemctl status snapd > before.txt
+  "$TESTSTOOLS"/snaps-state install-local test-snapd-journal-quota
+  snap set system experimental.quota-groups=true
+  tests.cleanup defer snap unset system experimental.quota-groups
+
+restore: |
+  # disable persistent journal
+  rm -rf /var/log/journal
+  systemctl kill --signal=SIGUSR1 systemd-journald
+    
+execute: |
+  echo "Wait for first boot to be done"
+  snap wait system seed.loaded
+
+  echo "Precondition check, persistent journal is not available by default"
+  not test -e /var/log/journal
+
+  echo "Check that persistent journal can be enabled"
+  snap set core journal.persistent=true
+  test -e /var/log/journal/
+  test -e /var/log/journal/.snapd-created
+
+  # Create a journal quota group and set arbitrary limits
+  echo "Create a journal namespace with the logger snap in it"
+  snap set-quota group-one --journal-size=16MB --journal-rate-limit=100/10ms test-snapd-journal-quota
+
+  # this check relies on *anything* getting logged; enabling persistent
+  # journal writes an entry about journal size, which should be sufficient.
+  MACHINE_ID=$(cat /etc/machine-id)
+  retry -n30 --wait 1 test -e "/var/log/journal/$MACHINE_ID"
```

Comment:
Please double check taht `-n30` really works, I think it should be `-n 30`

Reply:
That seems to be okay, I just tried:
```
$ ./tests/lib/external/snapd-testing-tools/tools/retry -n20 false
retry: command false failed with code 1
retry: next attempt in 1 second(s) (attempt 1 of 20)
...
```

Reply:
It works, but changing it none the less.